### PR TITLE
MuleSoft Cloudhub: first commit

### PR DESCRIPTION
**Pull request by averbn**
Change(s):
   - Added MuleSoft Cloudhub data connector

### CODE REVIEW

**Comment by v-sabiraj** [comment]
Hey @averbn, @vu-socprime, as in sample data the value for the fields lastModified_d, createdAt_d, event_timestamp_d, enabled_b, environmentId_g, id_g, isSystem_b, line_d, organizationId_g are given in **(" ")**, these fields are acting as a string, can you please check and update, thanks. 

image

**Review by v-sabiraj — CHANGES_REQUESTED** [review]

**Comment by averbn** [comment]
> Hey @averbn, @vu-socprime, as in sample data the value for the fields lastModified_d, createdAt_d, event_timestamp_d, enabled_b, environmentId_g, id_g, isSystem_b, line_d, organizationId_g are given in **(" ")**, these fields are acting as a string, can you please check and update, thanks.
> 
> image

The field types in the schema seem to be specified correctly:
image

The fields you specified are also not equal to " " in the real sample data:

image

File: Solutions/MuleSoftCloudhub/Data Connectors/MuleSoft_Cloudhub_API_FunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	"additionalRequirementBanner": ">This data connector depends on a parser based on a Kusto Function to work as expected [**MuleSoftCloudhub**](https://aka.ms/sentinel-MuleSoftCloudhub-parser) which is deployed with the Azure Sentinel Solution.",
+	"graphQueries": [{
+			"metricName": "Total data received",
+			"legend": "MuleSoft_Cloudhub_CL",
+			"baseQuery": "MuleSoft_Cloudhub_CL"
+		}
+	],
+	"sampleQueries": [{
+		"description": "MuleSoft Cloudhub Events - All Activities.",
+		"query": "MuleSoft_Cloudhub\n | sort by TimeGenerated desc"
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/MuleSoftCloudhub/Data Connectors/MuleSoft_Cloudhub_API_FunctionApp.json]: Done, thanks

**Review by VenkataPrathibha — CHANGES_REQUESTED** [review]

File: Solutions/MuleSoftCloudhub/Data Connectors/host.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        "isEnabled": true,
+        "excludedTypes": "Request"
+      }
+    }
+  },
+  "extensionBundle": {
+    "id": "Microsoft.Azure.Functions.ExtensionBundle",
+    "version": "[1.*, 2.0.0)"
+  }
+}
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/MuleSoftCloudhub/Data Connectors/host.json:15]: done

File: Solutions/MuleSoftCloudhub/Data Connectors/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# DO NOT include azure-functions-worker in this file
+# The Python Worker is managed by Azure Functions platform
+# Manually managing azure-functions-worker may cause unexpected issues
+
+azure-storage-file-share==12.3.0
+azure-functions
+requests
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/MuleSoftCloudhub/Data Connectors/requirements.txt:7]: done

File: Solutions/MuleSoftCloudhub/Data Connectors/MuleSoft_Cloudhub_API_FunctionApp.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		},
+		{
+			"title": "",
+			"description": "**2. Configure the Function App**\n\n1. In the Function App, select the Function App Name and select **Configuration**.\n2. In the **Application settings** tab, select ** New application setting**.\n3. Add each of the following application settings individually, with their respective string values (case-sensitive): \n\t\tMuleSoftEnvId\n\t\tMuleSoftAppName\n\t\tMuleSoftUsername\n\t\tMuleSoftPassword\n\t\tWorkspaceID\n\t\tWorkspaceKey\n\t\tlogAnalyticsUri (optional)\n> - Use logAnalyticsUri to override the log analytics API endpoint for dedicated cloud. For example, for public cloud, leave the value empty; for Azure GovUS cloud environment, specify the value in the following format: `https://<CustomerId>.ods.opinsights.azure.us`.\n4. Once all application settings have been entered, click **Save**."
+		}
+	]
+}
```
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/MuleSoftCloudhub/Data Connectors/MuleSoft_Cloudhub_API_FunctionApp.json:121]: done

[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/MuleSoftCloudhub/Data Connectors/MuleSoft_Cloudhub_API_FunctionApp.json:121]: done

**Comment by KateShapovalova** [comment]
@v-sabiraj, @VenkataPrathibha, done, please check.

**Review by VenkataPrathibha — APPROVED** [review]

File: Solutions/MuleSoftCloudhub/Data Connectors/azuredeploy_Connector_MuleSoft_Cloudhub_API_AzureFunction.json [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+                        "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=', toLower(variables('FunctionName')),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', toLower(variables('FunctionName'))), '2019-06-01').keys[0].value, ';EndpointSuffix=',toLower(variables('StorageSuffix')))]",
+                        "logAnalyticsUri": "[variables('LogAnaltyicsUri')]",
+                        "WorkspaceID": "[parameters('WorkspaceID')]",
+                        "WorkspaceKey": "[parameters('WorkspaceKey')]",
+                        "MuleSoftEnvId": "[parameters('MuleSoftEnvId')]",
+                        "MuleSoftAppName": "[parameters('MuleSoftAppName')]",
+                        "MuleSoftUsername": "[parameters('MuleSoftUsername')]",
+                        "MuleSoftPassword": "[parameters('MuleSoftPassword')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/averbn/azure_sentinel_data_connectors/raw/main/mulesoft-azure-sentinel-data-connector/MuleSoftCloudhubAPISentinelConn.zip?raw=true"
```
Comment (v-sabiraj) [inline Solutions/MuleSoftCloudhub/Data Connectors/azuredeploy_Connector_MuleSoft_Cloudhub_API_AzureFunction.json:167]: hey @averbn, please update the value of "WEBSITE_RUN_FROM_PACKAGE" with https://aka.ms/sentinel-MuleSoftCloudhubAPI-functionapp, thanks.

Comment (averbn) [reply Solutions/MuleSoftCloudhub/Data Connectors/azuredeploy_Connector_MuleSoft_Cloudhub_API_AzureFunction.json]: Sure

**Review by v-sabiraj — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**